src/status_im/utils/universal_links/core.cljs

```diff
@@ -81,8 +85,8 @@
       (navigation/navigate-to-cofx (assoc-in cofx [:db :contacts/identity] public-key) :profile nil))))
 
 (fx/defn handle-eip681 [cofx url]
-  {:dispatch-n [[:navigate-to :wallet-send-transaction]
-                [:wallet/fill-request-from-url url :deep-link]]})
+  (eip681/parse-uri-handler url :deep-link)
+  {:dispatch [:navigate-to :wallet-send-transaction]})
```

Comment:
im not sure i understand this one

Reply:
if should look like this
```
(fx/merge cofx
                 #(when-let [parsed-message (parse-uri url)]
                       #(resolve-ens-addresses % parsed-message :deep-link nil])
                  (navigation/navigate-to-cofx :wallet-send-transaction nil)
```

Reply:
@yenda Thoughts here? Didn't we decide to handle it the way it's currently handled since we're checking for a  EIP681 url earlier up in the universal links stack and didn't want to have to parse it twice at this stage in the code?

Reply:
let's wait for @yenda  but still, if you want to dispatch here it should be at least 
```
{:dispatch-n [[:parse-uri-handler ][:navigate-to :wallet-send-transaction]}
```

Reply:
@acolytec3 well since `parse-uri-handler` is parsing anyway calling it here will parse again. I think you need to handle the link and qr code cases because the context is different:
- with the qr code you are in a callback, so it makes sense to dispatch an event, so you can handle it in the re-frame context
- with the deep-link you are already in the re-frame context, so you don't have to dispatch an event

So only the qr code context should dispatch an event and the deep-link context should call `resolve-ens-addresses` directly.

I would recommend using @flexsurfer first proposition except that you should move the when-let inside the resolve-ens-addresses function.

Reply:
Got it.  Will be resolved along those lines in next commit.

Reply:
this is still wrong it won't work because resolve-ens-address will produce a map of effects, and it won't be used, you need to do something like this to merge the effects:

```
(fx/merge cofx
          (resolve-ens-addresses % url :deep-link nil]
          (navigation/navigate-to-cofx :wallet-send-transaction nil)
```

Reply:
Sorry for misunderstanding.  Should be resolved along these lines in the latest commit.